command/genesis/polybft_params.go

```diff
@@ -134,24 +140,71 @@ func (p *genesisParams) generatePolyBftChainConfig(o command.OutputFormatter) er
 		}
 	}
 
+	voteDelay, err := common.ParseUint256orHex(&p.voteDelay)
+	if err != nil {
+		return err
+	}
+
+	votingPeriod, err := common.ParseUint256orHex(&p.votingPeriod)
+	if err != nil {
+		return err
+	}
+
+	if votingPeriod.Cmp(big.NewInt(0)) == 0 {
+		return errInvalidVotingPeriod
+	}
+
+	proposalThreshold, err := common.ParseUint256orHex(&p.proposalThreshold)
+	if err != nil {
+		return err
+	}
+
+	governorAdminAddr := types.ZeroAddress
+	// if no admin is defined, zero address will be the owner,
+	// meaning no new proposers and executors besides genesis validator
+	// set can be added/removed later
+	if p.governorAdmin != "" {
```

Comment:
Is there a validation somewhere to make sure the provided address is valid?

Reply:
No. I don't know how a validation would look like, since` types.StringToAddress` always returns an address, plus I think we don't have any validation on address flags, except if they are nil or empty. In this case, we allow empty, but we will set the admin to a zero address in that case.

Reply:
`types.IsValidAddress(p.governorAdmin)`

Reply:
You can optionally consider using this function to validate an address:
types.go
`func IsValidAddress(address string) error`

Reply:
A user can provide any value and it will be converted to a "valid" address implicitly which is not good. The conversation here is just about internal usage of this value - type conversion. But the given address must be valid so everything is explicit for a user.

Reply:
Fix: https://github.com/0xPolygon/polygon-edge/pull/1519/commits/bb28c7adce1a3321da02389d78ac7ab537dc5df0